cEP-0035: Extending support for newer versions of Python

anshalshukla opened the pull request:
This proposal describes the approach that will be followed
to enable coala work with newer version of Python.

closes #191

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,52 @@
+| -------- | ------------------------------------- |
+| cEP      | 0035                                  |
+| Version  | 1.0                                   |
+| Title    | Extending support for newer version of Python  |
+| Authors  | Anshal Shukla <mailto:shukla.anshal85@gmail.com> |
```
Could improve the formatting here a bit.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,52 @@
+
+## Abstract
+
+This cEP describes the upgrades and steps required to enable coala to support newer versions of Python. This project has been taken up as a [GSoC](https://summerofcode.withgoogle.com/organizations/6490386047959040/#4634486723051520) project.
```
Mention the GSoC year, also this line is too long, please use a decent max line length.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,52 @@
+
+## Proposed Approach
+
+### 1. Keeping the CI/CD pipeline green
```
Let's not use numbers in headings.

abhishalya reviewed (changes requested):
This looks very vague in terms of the changes being made as a part of the project. It is important to pinpoint the changes as much as possible. A lot of which is already available in your project.

abhishalya commented:
Please no more commits like `aa`, `aaa`. It's an extremely bad practice to write such commits.

anshalshukla commented:
I had to change the commit message to "Add support for newer versions of Python" because having "Extending" makes the commit headlog imperative. coala throws an error with that. Let me know if I should change that project-wide.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,187 @@
+| -------- | ------------------------------------- |
+| cEP      | 0035                                  |
+| Version  | 1.0                                   |
+| Title    | Extending support for newer version of Python    |
+| Authors  | Anshal Shukla <mailto:shukla.anshal85@gmail.com> |
```
You can improve formatting a bit here.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,187 @@
+
+## Abstract
+
+This cEP describes the upgrades and steps required to enable coala to support newer versions of Python. This project has been taken up as a [GSoC'21](https://summerofcode.withgoogle.com/organizations/6490386047959040/#4634486723051520) project.
```
Group all links at the bottom and use them here.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,187 @@
+
+## Introduction
+
+coala, as of now, supports python up to version 3.6, which is quite old. Lack of support of newer versions of python has been a significant hurdle in the development of coala. Version 3.6 was released back in 2016, and python has made a considerable amount of progress since then. coala, on the other hand, has largely not been able to benefit from these upgrades because of the strict version limit ...
```
Respect line length please.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,187 @@
+
+## Objectives
+
+TThe project's primary goal is to enable coala to support newer versions of Python, but this will be attained in multiple smaller steps. The goal is not just to facilitate support for newer versions of Python but also to upgrade the coala to support more recent dependencies. Upgrades will be performed over multiple repositories as the dependencies across the project are closely interlinked. Will  ...
```
coala-bears.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,187 @@
+### Keeping the CI/CD pipeline green
+
+Throughout the project, it needs to be ensured that the CI/CD pipeline remains unbroken as it is crucial for running the test suite, which is necessary for PR to get merged. This can be ensured by making commits keeping into consideration the pipeline. And make changes over multiple repositories before merging, in case required to keep the pipeline running.
+Currently, the bears CI is broken, so I will start by fixing it.
```
Not required.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,187 @@
+
+### Keeping the CI/CD pipeline green
+
+Throughout the project, it needs to be ensured that the CI/CD pipeline remains unbroken as it is crucial for running the test suite, which is necessary for PR to get merged. This can be ensured by making commits keeping into consideration the pipeline. And make changes over multiple repositories before merging, in case required to keep the pipeline running.
```
cEP should only focus on the potential enhancements, and it doesn't have to revolve around the project.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,187 @@
+
+### Upgrade dependency_management system 
+
+The dependency_management system provides a Python API for familiar package managers. It provides a generic method to install dependencies in the host operating system. It has been a while since it has been upgraded. Thus it supports only the earlier versions of package managers. Will upgrade the Python, dependencies and versions of the package managers to access newer dependencies using the API. ...
```
Again same point as above, sentences like `will upgrade, will enable` etc do not seem fit for the cEP. This is basically just a plain list of enhancements, not the project proposal.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,187 @@
+
+### Upgrade bears
+
+Currently bears don't support newer pip versions, changes will be made to fix that. This is essential as while installing the latest version of a Python package might fail or install at a slower rate. This will be followed by upgrade in existing dependencies and Python version. Some changes might be required in mobans and coala as well to keep the pipeline up running. The problems associated with ...
```
Please improve the wordings as mentioned in previous points, not mentioning them again to reduce clutter. Just try to give factual info here.

abhishalya reviewed (changes requested):
This is looking much better than before, just take care of few comments mentioned inline. Also use a spell checker to ensure there are no typos or grammar errors.

abhishalya commented:
@anshalshukla Please make necessary changes, and rebase. This needs to be merged today.

anshalshukla commented:
@abhishalya Will do it soon.

anshalshukla commented:
@abhishalya, I guess this looks fine now?

abhishalya commented:
Resolve conflicts as well please.

abhishalya commented on cEP-0035.md:
````diff
@@ -0,0 +1,225 @@
+    # Python 3.7+
+    asyncio.run(main())
+    ```
+
````
Redundant line.

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,225 @@
+
+Lack of support for newer versions of Python has been a significant
+hurdle in the development of coala. coala, as of now, supports
+Python up to version 3.6, which is  quite old. Version 3.6 was
```
```suggestion
Python up to version 3.6, which is quite old. Version 3.6 was
```

abhishalya commented on cEP-0035.md:
```diff
@@ -0,0 +1,225 @@
+
+### Update Docker Image
+
+Fix docker image to serve latest version of coala and bears.
```
```suggestion
Fix docker image to serve the latest version of coala and bears.
```

anshalshukla committed:
cEP-0035: Add support for newer versions of Python

This proposal describes the approach to enable
coala to work with newer versions of Python.

Closes https://github.com/coala/cEPs/issues/191

anshalshukla commented:
@abhishalya Done!

abhishalya reviewed (dismissed):
Nice work :)

abhishalya commented:
@anshalshukla You didn't write valid commit short log. Please read the commit guidelines thoroughly.

```
closes coala#191
```

Please take care of these things before writing any commit. This PR is unmergable until everything is done properly.

anshalshukla commented:
@abhishalya Can I get this merged now.
